setup.py

```diff
@@ -1,10 +1,29 @@
-import os
 from setuptools import setup
-from setuptools.config import read_configuration
+import ast
+try:
+    from pathlib import Path
+except:
+    from pathlib2 import Path
```

Comment:
I would like to drop building on python 2 completely, since it is EOL, rather than use `os.path`. But if someone needs it, IMHO making him only to install `pathlib2` is less cruel than making him to rewrite half of the code.

Reply:
> I would like to drop building on python 2 completely

I thought we decided against that in kaitai-io/kaitai_struct#695?

> But if someone needs it, IMHO making him only to install `pathlib2` is less cruel than making him to rewrite half of the code.

Transitioning from Python 2 to 3 is besides the point here - we still support Python 2. The question is: when running on Python 2, what should our setup.py do:

* Use `pathlib2`, requiring Python 2 users to *manually* install an additional dependency, for minimally cleaner file manipulation code
* Use `open` and `os.path`, requiring no additional dependencies for users, at the cost of having to slightly rewrite two lines

I am very strongly for the second option. Requiring extra dependencies *to run setup.py* is very unusual, and I don't think it makes sense to break automatic package installation for all Python 2 users just so we have *slightly* cleaner code in our setup.py.

Reply:
* `pathlib` neither requires additional dependencies for python 3 users
* python 2 is EOL by now, why should we use `os.path` only to allow users of a deprecated dead version run `setup.py` without additional dependencies? They can always install a wheel, btw, or manually copy the file to the location where they need it.